### PR TITLE
Don't install in /usr/local/bin

### LINKED ISSUE #204: YVM fails to install, because it requires write permissions to /usr/local/bin
**Describe the bug**
The official installation instructions require write permissions to `/usr/local/bin`, and do not inform of it nor of any methods to override, thus making it impossible to install without downloading and inspecting/tweaking the installation file.

**To Reproduce**
1. Run official installation instructions: `curl -fsSL https://raw.githubusercontent.com/tophat/yvm/master/scripts/install.sh | bash`
2. See error:
```Installing Version: 2018-12-07T00-24-10.408Z
ln: failed to create symbolic link '/usr/local/bin/yvm': Permission denied
yvm failed to install, the command that failed was: https://github.com/tophat/yvm/blob/master/scripts/install.sh#L50
```

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Linux, docker image fedora:28
 - YVM Version: not installed, trying to install 2018-12-07T00-24-10.408Z
 - Shell: bash
 - Node version: v10.14.1

**Additional context**
Note that installing the files below $HOME, accessible to only one user, and symlinking that in /usr/local/bin, where it's accessible to all users, makes it problematic for those users, as it puts inaccessible files in their $PATH.

### PR DESCRIPTION
**Pull request by torinthiel**
Change installation script so that it does not install outside user home
directory (or whatever is passed as YMV_DIR/YVM_INSTALL_DIR).

## Description
Installation script:
- Removed code to install in /usr/local/bin
- source yvm directly from $YVM_DIR

### DevQA Steps
- Install yarn using new installation script in various situations. Note that write permission to `/usr/local/bin` is no longer required

### Comments:
- fixes #204

### CODE REVIEW

**Comment by jakebolam** [comment]
@francoiscampbell does this remove the need for: https://github.com/tophat/yvm/blob/master/src/yvm.sh#L62

File: scripts/install.sh [hunk 1]
```diff
@@ -75,5 +70,5 @@ if ! grep -q "${executable_source_string}" ~/.bash_profile; then
     echo ${executable_source_string} >> ~/.bash_profile
 fi
 
-echo "yvm successfully installed in ${YVM_DIR} with the 'yvm' command aliased as ${executable_alias_path}"
-echo "Open another terminal window to start using it, or type \"source /usr/local/bin/yvm\""
+echo "yvm successfully installed in ${YVM_DIR} with the 'yvm' command aliased as ${sh_install_path}"
```
Comment (jakebolam) [inline scripts/install.sh]: Possibly change sentence now (not aliased)
> yvm successfully installed in ${YVM_DIR} as ${sh_install_path}

**Comment by jakebolam** [comment]
Thanks @torinthiel for the contribution this looks great to me. 

Left a comment about removing the mention of aliases now, let me know.

File: scripts/install.sh (hunk 1 above)
Comment (torinthiel) [reply scripts/install.sh]: Makes sense, force-pushed with gix.

File: scripts/install.sh [hunk 2]
```diff
@@ -16,14 +16,10 @@ YVM_DIR=${YVM_INSTALL_DIR-"${HOME}/.yvm"}
 zip_download_path="${YVM_DIR}/yvm.zip"
 sh_install_path="${YVM_DIR}/yvm.sh"
 
-YVM_ALIAS_DIR=${YVM_ALIAS_DIR-"/usr/local/bin"}
-executable_alias_path="${YVM_ALIAS_DIR}/yvm"
 export_yvm_dir_string="export YVM_DIR=${YVM_DIR}"
-executable_source_string="source ${executable_alias_path}"
+executable_source_string="source ${sh_install_path}"
```
[the remark this answers is not in the document]
Comment (torinthiel) [reply scripts/install.sh]: That's going to be problematic, as we are dealing with several issues here:
- The source import itself, in both `~/.zshrc` and `~/.bash_profile`
- The already installed file in /usr/local/bin/yvm
- Possibility that user had it installed somewhere else, by supplying custom `$YVM_ALIAS_DIR` during installation
and one thing that I was going to raise after this one - the script should modify `~/.bashrc` and not `~/.bash_profile`

Although I kind of think that displaying anything on init is incorrect behaviour - this means that something will be displayed even for non-interactive sessions, where e.g. something is launching a command via `ssh` and interacting with it - in that case outputting anything will break the script. I've even seen a protection in some distros that will stop processing `~/.bashrc` for non-interactive shells.

[the remark this answers is not in the document]
Comment (torinthiel) [reply scripts/install.sh]: I think I could add a snippet that checks for yvm in `$PATH`, and if found and is a symlink `$YVM_DIR`, assumes it's leftovers from old installation - remove, and `sed`-out of `~/.zshrc`/`~/.bash_profile`, however there's a potential to remove something important for the user.

[the remark this answers is not in the document]
Comment (jakebolam) [reply scripts/install.sh]: I think you're right.

1. We should not output to the terminal.
2. We should address install in `.bashrc`

I think sourcing twice is favourable to deleting/detecting multiple sources of yvm

@torinthiel if you want to create issues for these, I'll merge this change

[the remark this answers is not in the document]
Comment (torinthiel) [reply scripts/install.sh]: Sure, I've created #225 and #226. I'll also create a coupe of others I have on my mind, willing to solve them as well.

[the remark this answers is not in the document]
Comment (torinthiel) [reply scripts/install.sh]: I've also pushed another commit, fixing the message from `yvm.js` and information in FAQ.

[the remark this answers is not in the document]
Comment (jakebolam) [reply scripts/install.sh]: Looks good, merging!

@torinthiel come join us in the slack channel `#yvm` in https://opensource.tophat.com/slack. I'm going to add you as a contributor so you have write access to the repo (and can push up branches directly)

[the remark this answers is not in the document]
Comment (torinthiel) [reply scripts/install.sh]: @jakebolam Thanks, accepted both. I'll start with #229, but I won't be disappointed if someone beats me doing it - given it's my first time with real code (And any node code at all), it's going to take a while.

**Pull request merged**